gamersnet_backend/routes/users/getAvatar.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+let fs = require('fs');
+
+let {getUserByID} = require('../../persistence/users');
+let {getAvatarByUserID} = require('../../persistence/avatar');
+
+async function getAvatar(request, response) {
+    let id = request.params.id
+    let result = null;
+
+    // for some reason, an invalid id parameter will be the string "undefined"
+    if (id !== 'undefined') {
+        result = await getUserByID(id);
+    }
+    
+    if (result === null) {
+        response.status(404).end();
+    } else {
+        result = await getAvatarByUserID(id);
```

Comment:
Are you using the same result variable for getting both user and Avatar?

Reply:
Turns out I am, while it does not seem to break anything, I'll change this as it can be considered a code smell